hicommonwealth/commonwealth
# Fixed bug where unauthenticated users would get subscription errors in the console.

## Link to Issue
Closes: 

- https://github.com/hicommonwealth/commonwealth/issues/8822
- https://github.com/hicommonwealth/commonwealth/issues/8821

## Description of Changes

- Moves the subscription toggle and all the associated logic of it to a dedicated class

## "How We Fixed It"


## Test Plan
- Load a thread when the user is not logged in. 
- Make sure there is at least one comment.  
- Then check logs and make sure theres no error.
- Make sure toggling of subscriptions works properly when on the thread page (and it has comments)

## Deployment Plan

1. 

## Other Considerations

-
